ui/panels-plugin/src/plugins/gauge-chart/GaugeChart.tsx

```diff
@@ -11,22 +11,262 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { AnyGraphQueryDefinition, useGraphQuery } from '@perses-ui/core';
+import * as echarts from 'echarts/core';
+import type { EChartsOption } from 'echarts';
+import { GaugeChart as EChartsGaugeChart } from 'echarts/charts';
 import {
-  JsonObject,
-  PanelProps,
-  AnyGraphQueryDefinition,
-} from '@perses-ui/core';
+  GridComponent,
+  DatasetComponent,
+  TitleComponent,
+} from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useMemo, useState, useLayoutEffect, useRef } from 'react';
+import { Box } from '@mui/material';
+import { CalculationsMap, CalculationType } from '../../model/calculations';
+import { formatValue, UnitOptions } from '../../model/units';
 
-export const GaugeChartKind = 'GaugeChart' as const;
+echarts.use([
+  EChartsGaugeChart,
+  GridComponent,
+  DatasetComponent,
+  TitleComponent,
+  CanvasRenderer,
+]);
 
-type GaugeChartKind = typeof GaugeChartKind;
-
-export type GaugeChartProps = PanelProps<GaugeChartKind, GaugeChartOptions>;
-
-interface GaugeChartOptions extends JsonObject {
+export interface GaugeChartProps {
   query: AnyGraphQueryDefinition;
+  width: number;
+  height: number;
+  calculation: CalculationType;
+  unit: UnitOptions;
 }
 
-export function GaugeChart(props: GaugeChartProps) {
-  return <div>{JSON.stringify(props)}</div>;
+// TODO (sjcobb): take threshold default colors from theme (green, red, yellow, gray)
+const thresholdColors = {
+  green: 'rgb(115, 191, 105)',
+  yellow: 'rgba(237, 129, 40, 0.89)',
+  red: 'rgba(245, 54, 54, 0.9)',
+  gray: '#e1e5e9',
+};
+
+const noDataOption = {
+  title: {
+    show: true,
+    textStyle: {
+      color: 'grey',
+      fontSize: 20,
+    },
+    text: 'No data',
+    left: 'center',
+    top: 'center',
+  },
+  xAxis: {
+    show: false,
+  },
+  yAxis: {
+    show: false,
+  },
+  series: [],
+};
+
+function GaugeChart(props: GaugeChartProps) {
+  const { query, width, height, calculation, unit } = props;
+  const { data } = useGraphQuery(query);
+
+  const option: EChartsOption = useMemo(() => {
+    // TODO (sjcobb): add loading spinner, share noDataOption with other charts
```

Comment:
Yeah, we're gonna have to figure out an approach to loading pretty soon. I was thinking maybe making use of MUI's `Skeleton` component.